sentry-compose/src/androidMain/kotlin/io/sentry/compose/SentryNavigationIntegration.kt

```diff
@@ -0,0 +1,52 @@
+package io.sentry.compose
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.NonRestartableComposable
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.navigation.NavController
+import androidx.navigation.NavHostController
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.android.navigation.SentryNavigationListener
+
+internal class SentryLifecycleObserver(
+    private val navController: NavController,
+    private val hub: IHub = HubAdapter.getInstance(),
+    private val navListener: NavController.OnDestinationChangedListener =
+        SentryNavigationListener(hub)
```

Comment:
Optional ctors as well or the overload annotation.

Reply:
mm I don't think it's needed here, cause this class is internal and is only supposed to be used in kotlin-only codebases, since compose is a kotlin framework

Reply:
So the problem is that, if they look at optional params, they will think, do I need a `hub`? do I need a `navListener`? even with default parameters, ideally, they'd not even think about it, that's why I believe having fewer options the better, we just need everything to make it testable.

Reply:
But they are not going to use this class at all, because it's marked as internal :) it's only us who's using this class